src/core/operations/Enigma.mjs

```diff
@@ -0,0 +1,199 @@
+/**
+ * Emulation of the Enigma machine.
+ *
+ * @author s2224834
+ * @copyright Crown Copyright 2019
+ * @license Apache-2.0
+ */
+
+import Operation from "../Operation";
+import OperationError from "../errors/OperationError";
+import * as Enigma from "../lib/Enigma";
+
+/**
+ * Enigma operation
+ */
+class EnigmaOp extends Operation {
+    /**
+     * Enigma constructor
+     */
+    constructor() {
+        super();
+
+        this.name = "Enigma";
+        this.module = "Default";
```

Comment:
We could consider creating a new module to put Enigma, Typex and any future related ops into. At the moment the codebase is nice and compact so it's not a problem, but we should bear it in mind for the future.